org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/WorkspaceHelper.java

```diff
@@ -58,10 +69,224 @@ public static IWorkspaceRoot getWorkspaceRoot() {
 
 	public static void delete(IProject project) {
 		try {
-			project.delete(true, new NullProgressMonitor());
```

Comment:
Do you know why IProject#delete(..) doesn't work ?

Reply:
I think some job/thread creates a file/folder in the project tree while we delete it.
I have seen the following error in the log: "TestProject/bin already exists". We sometimes do not delete a project properly.